test/lint.js

```diff
@@ -61,43 +100,77 @@ function checkSchema(dataFilename) {
-      // (see https://github.com/atlassian/better-ajv-errors/pull/21)
-      // Other issues with better-ajv-errors:
-      // - it feels better for performance to output messages one by one rather than a list
-      // - it seems to be losing some errors when output a list
-      ajv.errors.forEach(function(error) {
-        var message = betterAjvErrors(schema, data, [error], { indent: 2 });
-
-        console.log('\n    ' + message.replace(/\n/g, '\n    '));
+      console.error(`\x1b[31m  JSON Schema – ${ajv.errors.length} error(s)\x1b[0m`)
+      ajv.errors.forEach(e => {
+        // Output messages by one since better-ajv-errors wrongly joins messages
+        // (see https://github.com/atlassian/better-ajv-errors/pull/21)
+        // Other issues with better-ajv-errors:
+        // - it feels better for performance to output messages one by one rather than a list
+        // - it seems to be losing some errors when output a list
+        console.error(betterAjvErrors(schema, data, [e], {indent: 2}));
       });
+      return true;
     }
   }
 }
 
-dictPaths.forEach(function(dir) {
-  var absDir = path.resolve(path.join(__dirname, '..', dir));
+dictPaths.forEach(dir => {
+  const absDir = path.resolve(path.join(__dirname, '..', dir));
 
-  fs.readdirSync(absDir).forEach(function(filename) {
+  fs.readdirSync(absDir).forEach(filename => {
     if (path.extname(filename) === '.json') {
-      var absFilename = path.join(absDir, filename);
+      let hasSyntaxErrors = false,
+        hasSchemaErrors = false,
+        hasStyleErrors = false;
 
-      console.log(dir + '/' + filename);
+      const absFilename = path.join(absDir, filename);
+      const relativeFilePath = path.relative(process.cwd(), absFilename);
 
-      checkStyle(absFilename)
-      checkSchema(absFilename);
+      const spinner = ora({
+        stream: process.stdout,
+        text: relativeFilePath
+      });
 
-      console.log();
+      const console_error = console.error;
+      console.error = (...args) => {
+        spinner.stream = process.stderr;
+        spinner.fail(relativeFilePath);
+        console.error = console_error;
+        console.error(...args);
+      }
+
+      try {
+        hasStyleErrors = testStyle(absFilename)
+        hasSchemaErrors = testSchema(absFilename);
+      } catch (e) {
+        hasSyntaxErrors = true;
+        console.error(e);
+      }
+
+      if (hasSyntaxErrors || hasSchemaErrors || hasStyleErrors) {
+        hasErrors = true;
+        filesWithErrors.set(relativeFilePath, absFilename);
+      } else {
+        console.error = console_error;
+        spinner.succeed();
+      }
     }
   });
 });
 
 if (hasErrors) {
+  console.warn("");
+  console.warn(`Problems in ${filesWithErrors.size} file${filesWithErrors.size > 1 ? 's' : ''}:`);
+  for (const [fileName, file] of filesWithErrors) {
+    console.warn(fileName);
+    try {
+      testSchema(file);
```

Comment:
Looks like a double checking of files and message output.

Reply:
Again, this is from https://github.com/mdn/browser-compat-data/blob/9d92e6d7f88cfa4400423a39183c321d7a68c410/test/lint.js#L113-L132.